doc/nvim-tree-lua.txt

```diff
@@ -58,6 +58,107 @@ Print clipboard content for both cut and copy
 Resize the NvimTree window to the given size. Example: `:NvimTreeresize 50`
 resizes the window to the width of 50.
 
+==============================================================================
+SETUP                                *nvim-tree-setup*
+
+To configure the tree (and make it runnable), you should call the setup
+function.
+
+>
+    require'nvim-tree'.setup {
+      disable_netrw       = true,
+      hijack_netrw        = true,
+      open_on_setup       = false,
+      ignore_ft_on_setup = {},
+      auto_close          = false,
+      tab_open            = false,
+      hijack_cursor       = false,
+      update_cwd          = false,
+      update_focused_file = {
+        enable      = false,
+        update_cwd  = false,
+        ignore_list = {}
+      },
+      system_open = {
+        cmd  = nil,
+        args = {}
+      },
+    }
+<
+
+As options are currently being migrated, configuration of global options in 
+*nvim-tree-options* should be done BEFORE the setup call.
+
+Here is a list of the options available in the setup call:
+- |disable_netrw|: completely disable netrw
+  type: `boolean`
+  default: `true`
+
+- |hijack_netrw|: hijack netrw windows (overriden if |disable_netrw| is `true`)
+  type: `boolean`
+  default: `true`
+
+- |open_on_setup|: will automatically the tree when running setup if current
+  buffer is a directory, is empty or is unnamed.
+  type: `boolean`
+  default: `false`
+
+- |ignore_ft_on_setup|: list of filetypes that will make |open_on_setup| not
+  open. You can use this option if you don't want the tree to open in some
+  scenarios (eg using vim startify).
+  type: `{string}`
+  default: `{}`
+
+- |auto_close|: force closing neovim when the tree is the last window in the view.
+  type: `boolean`
+  default: `false`
+
+- |tab_open|: opens the tree automatically when switching tabpage or opening a new
```

Comment:
or even
```lua
auto_open = {
  startup = bool,
  tab = bool,
}
```
?

Reply:
they are not really related. auto_open will open on startup automatically, tab will only keep the window open as if it was the same screen. I guess i'd like to rename the tab_open one because it's not really explicit.